src/interceptors/fetch/index.ts

```diff
@@ -70,6 +70,14 @@ export class FetchInterceptor extends Interceptor<HttpRequestEventMap> {
 
       if (mockedResponse) {
         this.log('received mocked response:', mockedResponse)
+        
+        if (request.signal.aborted) {
+          this.log('request aborted, ignoring mocked response');
+
+          // Throws an "AbortError" DOMException
+          await pureFetch(request)
```

Comment:
We could throw a `DOMException` manually here, but I wanted the error message to have the same wording as the original. Calling `fetch` here throws that error.

Reply:
I think it's better to rely on whichever `fetch` the consumer is using here. For example, I doubt that global fetch in Node will throw a `DOMException` 😉 So we remain consistent in this regard. 

My only concern with this is that we're creating a special clause for the aborted requests within the mocked response handling. I think it's confusing. If I understood it right, we will know about `request.signal.aborted` by the time we receive a mocked response (or don't receive any). With that in mind, would this produce the same result:

```js
if (mockedResponse && !request.signal?.aborted) {
  // ...this closure has an early return
}

// This one handled all the other use cases,
// which includes the lack of mocked response or
// aborted request regardless of the mocked response state.
pureFetch(...) // the same as now.
```

Basically, instead of treating aborted requests as a special case, we instead instruct the interceptor to only handle mocked responses if the request hasn't been aborted. This makes sense semantically because when the client aborts a request, it mustn't expect the mocked response to be sent. 

What do you think about this?

Reply:
Yes, this is way better. 

> Also, isn't request.signal potentially undefined?

According to the typings it is not. I have tested Safari, Chrome, Firefox and Deno and you always get an `AbortSignal`. The spec says ["A Request object has an associated signal (null or an AbortSignal object), initially null."](https://fetch.spec.whatwg.org/#request-signal) though, so I think assuming it might be null might be the safest bet.

Reply:
Thanks for double-checking this.